code/scripting/api/objs/shipwepselect.cpp

```diff
@@ -119,6 +143,137 @@ ADE_VIRTVAR(ShipClassIndex,
 	return ade_set_args(L, "i", (current.getSlot()->original_ship_class + 1));
 }
 
+//**********HANDLE: loadout amount
+ADE_OBJ(l_Loadout_Amount, wss_unit_count_h, "loadout_amount", "Loadout handle");
+
+ADE_INDEXER(l_Loadout_Amount,
+	"number idx",
+	"Array of ship bank weapons. 1-3 are Primary weapons. 4-7 are Secondary weapons. Note that banks "
+	"that do not exist on the ship class are still valid here but should hold nil data. Also note that "
+	"primary banks will also hold the value of 1 even if it is ballistic.",
+	"number",
+	"Amount of the currently loaded weapon, -1 if bank has no weapon, or nil if the ship or index is invalid")
+{
+	wss_unit_count_h current;
+	int idx = -1;
+	if (!ade_get_args(L, "oi", l_Loadout_Amount.Get(&current), &idx))
+		return ADE_RETURN_NIL;
+
+	// Will eventually be writable so that Lua can manipulate the player loadout!
+	// Note may need to do some checking that the bank is valid for the current ship
+	// and that the amount-to-write is the correct size for the weapon/ship cargo size
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	if (idx < 1 || idx > 7) {
+		return ADE_RETURN_NIL;
+	};
+	idx--; // Convert to Lua's 1 based index system
+
+	return ade_set_args(L, "i", current.getBank()->wep_count[idx]);
+}
+
+ADE_FUNC(__len, l_Loadout_Amount, nullptr, "The number of weapon banks in the slot", "number", "The number of banks.")
+{
+	return ade_set_args(L, "i", MAX_SHIP_PRIMARY_BANKS + MAX_SHIP_SECONDARY_BANKS);
```

Comment:
```suggestion
	return ade_set_args(L, "i", MAX_SHIP_WEAPONS);
```
MAX_SHIP_WEAPONS is defined as the sum of PRIMARY_BANKS and SECONDARY_BANKS